js/app.js

```diff
@@ -1 +1,179 @@
+    console.log("set to computer prints user");
+    let userName = document.getElementById("userName").value;
+    document.getElementById("user-display").textContent = `${userName} is playing`;
+  } else if (whoIsThePlayer === "user") {
+    console.log("set to user prints computer");
+    document.getElementById("user-display").textContent = "computer is playing";
+  } else {
+    console.log("neither is playing");
+  }
+}
+
+// to win you must hit one of these combinations
+function checkWinner(arr /*has to be 3 x 3*/) {
+  //diagonal win
+  if (arr[0][0] === arr[1][1] && arr[1][1] === arr[2][2]) {
+    return arr[0][0];
+  }
+  if (arr[0][2] === arr[1][1] && arr[1][1] === arr[2][0]) {
+    return arr[0][2];
+  }
+
+  // first column win
+  if (arr[0][0] === arr[0][1] && arr[0][1] === arr[0][2]) {
+    return arr[0][0];
+  }
+  // second column win
+  if (arr[1][0] === arr[1][1] && arr[1][1] === arr[1][2]) {
+    return arr[1][0];
+  }
+  // third column win
+  if (arr[2][0] === arr[2][1] && arr[2][1] === arr[2][2]) {
+    return arr[2][0];
+  }
+
+  // first row win
+  if (arr[0][0] === arr[1][0] && arr[1][0] === arr[2][0]) {
+    return arr[0][0];
+  }
+  //second row win
+  if (arr[0][1] === arr[1][1] && arr[1][1] === arr[2][1]) {
+    return arr[0][1];
+  }
+  // third row win
+  if (arr[0][2] === arr[1][2] && arr[1][2] === arr[2][2]) {
+    return arr[0][2];
+  }
+  return false;
+}
+
+// empty board for user and computer moves to occupy
+const theBoard = [
+  ["", "", ""],
+  ["", "", ""],
+  ["", "", ""]
+];
+// use to interpolate moves without storing each individual unit
+const names = ["zero", "one", "two"]
+
+// i is x
+// j is y
+function redraw(board) {
+  for (let i = 0; i < 3; i++) {
+    for (let j = 0; j < 3; j++) {
+      // placing moves on board
+      document.getElementById(`${names[i]}-${names[j]}`).textContent = board[i][j];
+    }
+  }
+}
+
+
+function playMove(x, y, mark) {
+  // this checks if the move has been played already. any value other than 0 will result in a false and alert from theGame function
+  if (theBoard[x][y].length !== 0) {
+    return false;
+  }
+  theBoard[x][y] = mark;
+  return true;
+}
+
+// run checkWinner function after user and computer plays, then returns winner to results div when conditions are met
+function runCheckWinner(whoIsThePlayer, board) {
+  if (checkWinner(board)) {
+    // changes value to true to trigger the game to stop
+    isGameWon = true;
+    // user won
+    if (whoIsThePlayer === "user") {
+      let userName = document.getElementById("userName").value;
+      document.getElementById("results").textContent = `wowie ${userName}, you won :)`;
+      return true;
+      // computer wins
+    } else if (whoIsThePlayer === "computer") {
+      document.getElementById("results").textContent = `the computer won :(`;
+      return true;
+    }
+  }
+}
+
+// recursive function playValue returns eventListener callback function
+function playValue(x, y) {
```

Comment:
I wanted to mention: when you were talking about this function earlier, you mentioned the word "recursive."  I wanted to note that a recursive function is a function which calls itself.  So this is not a recursive function.  It is a function which returns a function though.  You could also do some reading about the distinction between first class functions versus higher order functions.  It's a little bit more on the pedantic side of things but can be useful (again) in job interviews.